client/dimensions_api.go

```diff
@@ -92,25 +93,25 @@ func (api DatasetAPI) GetDimensions(ctx context.Context, instanceID string) ([]*
 }
 
 // PatchDimensionOption make a HTTP patch request to update the node_id and/or order of the specified dimension.
-func (api DatasetAPI) PatchDimensionOption(ctx context.Context, instanceID string, d *model.Dimension, order *int) error {
+func (api DatasetAPI) PatchDimensionOption(ctx context.Context, instanceID string, d *model.Dimension, order *int) (string, error) {
 	if len(instanceID) == 0 {
-		return ErrInstanceIDEmpty
+		return headers.IfMatchAnyETag, ErrInstanceIDEmpty
 	}
 	if d == nil {
-		return ErrDimensionNil
+		return headers.IfMatchAnyETag, ErrDimensionNil
 	}
 	dim := d.DbModel()
 	if dim == nil {
-		return ErrDimensionNil
+		return headers.IfMatchAnyETag, ErrDimensionNil
```

Comment:
I think this should be "" (but it's not very important, as this value will be ignored because it's an error)